src/transaction/transaction.rs

```diff
@@ -75,6 +80,34 @@ impl Transaction {
             .await
     }
 
+    /// Pessimistically lock and get the value associated with the given key.
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{Value, Config, transaction::Client};
+    /// # use futures::prelude::*;
+    /// # futures::executor::block_on(async {
+    /// # let client = Client::new(Config::new(vec!["192.168.0.100", "192.168.0.101"])).await.unwrap();
+    /// let mut txn = client.begin_pessimistic().await.unwrap();
+    /// let key = "TiKV".to_owned();
+    /// let result: Option<Value> = txn.get_and_lock(key).await.unwrap();
+    /// // now the key "TiKV" is locked, other transactions cannot modify it
+    /// // Finish the transaction...
+    /// txn.commit().await.unwrap();
+    /// # });
+    /// ```
+    pub async fn get_and_lock(&mut self, key: impl Into<Key>) -> Result<Option<Value>> {
```

Comment:
Is it more correct to call it "lock_and_get"?

Reply:
Another name I've considered is `get_for_update`.

Reply:
Sounds good